jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceListing.java

```diff
@@ -84,12 +94,12 @@ public static String getAsHTML(Resource resource, String base, boolean parent, S
         }
 
         // Perform sort
-        Comparator<? super Resource> sort = switch (sortColumn)
-        {
-            case "M" -> ResourceCollators.byLastModified(sortOrderAscending);
-            case "S" -> ResourceCollators.bySize(sortOrderAscending);
-            default -> ResourceCollators.byName(sortOrderAscending);
-        };
+        Comparator<? super Path> sort = switch (sortColumn)
+            {
+                case "M" -> PathCollators.byLastModified(sortOrderAscending);
+                case "S" -> PathCollators.bySize(sortOrderAscending);
+                default -> PathCollators.byName(sortOrderAscending); // sorts by filename only (not full path)
+            };
         listing.sort(sort);
```

Comment:
`Resource.list()` has already done a sort by the natural order.   We should not double sort.  Either list returns unsorted OR we optionally pass in a comparator to list.

Reply:
The API for `Resource.list()` specifically states "Ordering is unspecified".

https://github.com/eclipse/jetty.project/blob/1c2290dca7a553a33933f33580c66cc0ac187925/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java#L265-L276